chia/server/server.py

```diff
@@ -639,6 +639,14 @@ async def wrapped_coroutine() -> Optional[Message]:
                     if response is not None:
                         response_message = Message(response.type, full_message.id, response.data)
                         await connection.send_message(response_message)
+                    # check that this call needs a reply
+                    elif metadata.reply_types is not None and len(metadata.reply_types) > 0:
```

Comment:
@arvidn this is where we catch the failed request and return a failure message instead of just not responding

Reply:
So the `.reply_types` aren't actually used anywhere else.  In https://github.com/Chia-Network/chia-blockchain/pull/13565 I refactored from multiple api decorators, one for each piece of information, to a single decorator.  While doing so I noticed that the information from the reply type decoration wasn't used.  It seemed like it should be used instead of the separately defined mapping.

https://github.com/Chia-Network/chia-blockchain/blob/f9f74481713b0edce85fcc6dbb000fe840b2a7db/chia/protocols/protocol_state_machine.py#L14-L45

But...  I hadn't gotten back to refactoring that duplication/dead code.  I hesitate to use `.reply_types` in the mean time since it may not be in sync with the other definition of the data.

Reply:
asserting the correct message type is done here

https://github.com/Chia-Network/chia-blockchain/blob/4db7fa956c72d9295a2d58b0ddf7651b33ed93af/chia/server/ws_connection.py#L366